presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -126,6 +132,12 @@
     private final boolean skipTargetCleanupOnRollback;
     private final boolean undoMetastoreOperationsEnabled;
 
+    private static Supplier<ScheduledExecutorService> leaseRenewalExecutor = Suppliers.memoize(() ->
+            Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder()
```

Comment:
So we have one thread for all the queries. What happens if the lease-renewal latency spiked for some reason and bunch of requests queued up and the partition leases of the queued ones expire before their turn comes ? just trying to understand under heavy load/ edge cases.

Reply:
This is an interesting point. Although there is a theoretical possibility of the lease expiring intermittently, the impact is likely to be minimal. Is it worth introducing a more robust thread pool to handle that situation?

Reply:
Do we have an estimate how many lease-renewal requests might happen per sec ? Since lease-duration is 0 by default, do we have an estimate as to how many queries will make use of this new param ? just trying to gauge the scale.

Reply:
> Since lease-duration is 0 by default, do we have an estimate as to how many queries will make use of this new param

If lease duration is set to 0, the partition leasing is not used at all. So this doesn't apply. It is not useful for short interactive queries. I am coming from the perspective of long-running Presto on Spark jobs.